src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -807,11 +807,35 @@ str2severity (const char *str)
   return RPM_OSTREE_ADVISORY_SEVERITY_NONE;
 }
 
+static int
+compare_advisory_refs (gconstpointer ap,
+                       gconstpointer bp)
+{
+  /* We use URLs to sort here; often CVE RHBZs will have multiple duplicates for each stream
+   * affected (e.g. Fedora, EPEL, RHEL, etc...), but we want the first one, which contains
+   * all the juicy details. A naive strcmp() sort on the URL gives us this. */
+  DnfAdvisoryRef *a = *((DnfAdvisoryRef**)ap);
+  DnfAdvisoryRef *b = *((DnfAdvisoryRef**)bp);
+  return strcmp (dnf_advisoryref_get_url (a), dnf_advisoryref_get_url (b));
```

Comment:
This is missing the case where any one of `a`, `b`, `url_a` or `url_b` are NULL.